assets/js/phoenix_live_view.js

```diff
@@ -511,11 +511,36 @@ export class LiveSocket {
     for(let type of ["change", "input"]){
       this.on(type, e => {
         let input = e.target
-        let key = input.type === "checkbox" ? "checked" : "value"
-        if(this.prevInput === input && this.prevValue === input[key]){ return }
+
+        let key
```

Comment:
Hey @alexgaribay what's up 👋 

Tested here with a multiple select input and it worked like a charm 🎉 .

Just one suggestion: I think we could get the input value as soon as we know its type so we don't need this key variable anymore, something like:

```JS
let value
if (input.type === "select-multiple") {
  value = Array.from(input.selectedOptions).map(option => option.value)
} else
  value = input.value
}
```

Not sure if this `Array.from` and `map` cause some issue here though, but it's easier to read :see_no_evil:

Reply:
I like the idea.  fyi on Array.from - https://github.com/phoenixframework/phoenix_live_view/pull/182

Reply:
@feliperenan I condensed to the code to follow the suggestion.

Reply:
Nice 👍 

Just to know, is there some reason why you prefer the old `for` loop instead of `Array.from` and map? maybe because my approach is more expensive 🤔 

Anyway, we also could move this logic to a specific function, something like this:

```JS
let value = this.getValuesFromInput(input)

getValuesFromInput(input) {
  if (input.type === "checkbox") {
    return input["checked"]
  }
  else if (input.type === "select-multiple") {
    let selectedOptions = input["selectedOptions"]
    let values = []
    for (let i = 0; i < selectedOptions.length; i++) {
      values.push(selectedOptions.item(i).value)
    }
    // Convert to a string for easier array comparison
    return JSON.stringify(values)
  }
  else {
    return input["value"]
  }
}
```

Wdyt?

Reply:
Ok, maybe because it's not IE11 compatible :/